common/lib/types/message.js

```diff
@@ -187,22 +187,30 @@ var Message = (function() {
 								throw new Error('Unable to decrypt message; not an encrypted channel');
 							}
 						case 'vcdiff':
-							if(context.plugins && context.plugins.vcdiff) {
-								try {
-									var deltaBase = context.baseEncodedPreviousPayload;
-									if(typeof deltaBase === 'string') {
-										deltaBase = BufferUtils.utf8Encode(deltaBase);
-									}
-
-									data = BufferUtils.typedArrayToBuffer(context.plugins.vcdiff.decode(data, deltaBase));
-									lastPayload = data;
-								} catch(e) {
-									throw new ErrorInfo('Vcdiff delta decode failed with ' + e, 40018, 400);
+							if(!context.plugins || !context.plugins.vcdiff) {
+								throw new ErrorInfo('Missing Vcdiff decoder (https://github.com/ably-forks/vcdiff-decoder)', 40019, 400);
+							}
+							if(typeof Uint8Array === 'undefined') {
+								throw new ErrorInfo('Delta decoding not supported on this browser (need ArrayBuffer & Uint8Array)', 40020, 400);
+							}
+							try {
+								var deltaBase = context.baseEncodedPreviousPayload;
+								if(typeof deltaBase === 'string') {
+									deltaBase = BufferUtils.utf8Encode(deltaBase);
 								}
-								continue;
-							} else {
-								throw new Error('Missing Vcdiff decoder (https://github.com/ably-forks/vcdiff-decoder)');
+
+								/* vcdiff expects Uint8Arrays, can't copy with ArrayBuffers. (also, if we
```

Comment:
This provided me food for thought. Nothing to stop us merging now but, FWIW: https://github.com/ably/ably-js/issues/659